src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -227,6 +230,63 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             }
         }
 
+        public override BoundNode VisitLValuePlaceholder(BoundLValuePlaceholder node)
+        {
+            return PlaceholderReplacement(node);
+        }
+
+        public override BoundNode VisitRValuePlaceholder(BoundRValuePlaceholder node)
+        {
+            return PlaceholderReplacement(node);
+        }
+
+        /// <summary>
+        /// Returns substitution currently used by the rewriter for a placeholder node.
+        /// Each occurrence of the placeholder node is replaced with the node returned.
+        /// Throws if there is no substitution.
+        /// </summary>
+        private BoundExpression PlaceholderReplacement(BoundValuePlaceholderBase placeholder)
+        {
+            var value = _placeholderReplacementMapDoNotUseDirectly[placeholder];
+            AssertPlaceholderReplacement(placeholder, value);
+            return value;
+        }
+
+        [Conditional("DEBUG")]
+        private static void AssertPlaceholderReplacement(BoundValuePlaceholderBase placeholder, BoundExpression value)
+        {
+            Debug.Assert(value.Type.Equals(placeholder.Type, ignoreCustomModifiersAndArraySizesAndLowerBounds: true));
```

Comment:
I think it is fine to ignore dynamic as well. VB doesn't have dynamic that is why it didn't mention it.